### DISCUSSION TITLE
Way to setup a global git town config?

### DISCUSSION THREAD

**Discussion by viniciusth** [git-town/git-town#4111] [General]
Hey, I work in a company with many many repo's, and it gets pretty annoying having to run `git town config setup` in every single repo, specially since I have to go fetch my github API token again, it would be great to have a way to setup a few things like the github token in a global manner

**Comment by viniciusth** [comment]
Just realized that in the docs if you go into the config sections it shows a way to set it as a global git config, maybe make this a y/n when using `git town config setup` or adding a `--global` flag to the command?

**Reply by kevgo** [reply]
Thanks for the great feedback. If you have a lot of repos, copy-and-pasting the [configuration file](https://www.git-town.com/configuration-file) into each repo should make setup easier. 

I like the idea of asking the user whether to set up the access keys globally. I have created https://github.com/git-town/git-town/issues/4112 to track the implementation of this idea.

**Comment by WhosNickDoglio** [comment]
Apologies for resurrecting a very old discussion, I"m happy to create a new one or a new issue _but_ I'd really like to set most (preferably **all**) git town configurations using `git config [--global] git-town`. 

Sometimes I have to jump into a new repo for some quick work and it's a litle annoying have to run through the setup wizard for some basic stuff just trying to run a `git town hack`, etc.

Right now it seems to require re-setting the following: 
1. git aliases
2. Main branch 
3. Origin hostname
4. Forge type
5. Configuration storage

The git aliases are already setup in my global `.gitconfig` so I'm not sure why these should pop up every time. I do _believe_ the rest could be auto detected or at the very least after I set a default value in my global git town config I should no longer get prompted about them.

**Reply by kevgo** [reply]
Thanks for bringing this up. I agree that the setup assistant needs some fine-tuning. Currently it behaves like this:

- If the codebase already contains a `git-town.toml`, it asks only for data that isn't in that file (see [this e2e test](https://github.com/git-town/git-town/blob/main/features/init/existing_configuration/config_file_complete.feature)). Settings in the config file apply to everybody working on that codebase, so no point in asking about them again.
- If global Git config exists on the local machine, it still asks for all config options, but pre-populates dialogs with the existing settings. If you leave those unchanged, we don't write local Git config so that the global setting remains effective. See [this e2e test](https://github.com/git-town/git-town/blob/main/features/init/existing_configuration/git_global.feature). The intent here is to make it easy to override global defaults on a per-repo basis.

I agree the current UX can be noisy if you're *not* trying to override anything, and just want to configure what's missing for a new repo. How can we support this use case better?

1. By default, the setup assistant should ask only for data not already configured anywhere (global Git metadata, config file, environment variables). This is for people who want to reuse all existing configuration for a new repo (most use cases).
2. Add a CLI flag like `git town init --all` that makes the setup assistant ask for all values, even when they are already configured. This is for users who want to override existing configuration for a new repo.

Would that address your use case, or are you envisioning something different?

**Reply by WhosNickDoglio** [reply]
> Would that address your use case, or are you envisioning something different?

Yes would be perfect!
